### PR TITLE
add config values to json and mfg attachment

### PR DESCRIPTION
**Pull request by wallacbe**

### CODE REVIEW

**Comment by wallacbe** [comment]
https://github.com/google/openhtf/issues/141

File: openhtf/io/output/json_factory.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ def default(self, obj):
 
   # pylint: disable=invalid-name
   def __call__(self, test_record):
+    test_record.metadata['config'] = conf._asdict()
```
Comment (grybmadsci) [inline openhtf/io/output/json_factory.py]: er, i wouldn't do it only here, i'd put it in the test record metadata itself

**Comment by wallacbe** [comment]
made change, PTAL

File: openhtf/exe/test_state.py [hunk 2]
```diff
@@ -63,6 +64,7 @@ def __init__(self, test_data, plug_manager, dut_id, station_id):
     self.record = test_record.TestRecord(
         dut_id=dut_id, station_id=station_id, code_info=test_data.code_info,
         metadata=test_data.metadata)
+    self.record.metadata['config'] = conf._asdict()
```
Comment (grybmadsci) [inline openhtf/exe/test_state.py]: can you initialize it when the metadata gets initialized, i believe upon creation of the test data object.

**Comment by wallacbe** [comment]
Made change, PTAL

File: openhtf/io/output/mfg_inspector.py [hunk 3]
```diff
@@ -109,6 +109,10 @@ def _PopulateHeader(record, testrun):
     testrun_code = testrun.failure_codes.add()
     testrun_code.code = details.code
     testrun_code.details = details.description
+  if 'config' in record.metadata:
+   attachment = testrun.info_parameters.add()
```
Comment (grybmadsci) [inline openhtf/io/output/mfg_inspector.py]: two space indent, looks like this is only one

**Comment by grybmadsci** [comment]
LGTM, fix indent and rebase and i'll get this merged in :)

**Comment by wallacbe** [comment]
Rebased, fixed unit test, PTAL

**Pull request merged**